File: templates/web-java-selenide/src/test/java/web/pages/HotelsPage.java
package web.pages;

import com.codeborne.selenide.ElementsCollection;
import com.codeborne.selenide.SelenideElement;
import io.qameta.allure.Step;

import static com.codeborne.selenide.Condition.text;
import static com.codeborne.selenide.Selenide.$;
import static com.codeborne.selenide.Selenide.$$;
import static com.codeborne.selenide.Selenide.open;

/** Каталог отелей (авторизованная зона). */
public class HotelsPage {

    // h1 на странице нет; названия карточек — тоже h2, поэтому ищем заголовок по тексту
    public final SelenideElement heading = $$("h2").findBy(text("Найдите идеальный отель"));
    public final SelenideElement searchInput = $("#hotel-search-input");
    public final ElementsCollection hotelNames = $$("[id^=hotel-card-name-]");

    @Step("Открываем каталог отелей")
    public HotelsPage openPage() {
        open("/hotels");
        return this;
    }
}

File: templates/web-python-playwright/pages/hotels_page.py
"""Каталог отелей (авторизованная зона)."""
import allure
from playwright.sync_api import Locator

from pages.base_page import BasePage


class HotelsPage(BasePage):
    path = "/hotels"

    @property
    def heading(self) -> Locator:
        return self.page.get_by_role("heading", name="Найдите идеальный отель")

    @property
    def search_input(self) -> Locator:
        return self.page.locator("#hotel-search-input")

    @property
    def hotel_cards(self) -> Locator:
        # Контейнер карточки — div с id вида hotel-card-<id>; вложенные элементы
        # с тем же префиксом (name, кнопки) — другие теги, div-фильтра достаточно
        return self.page.locator("div[id^=hotel-card-]")

    @property
    def hotel_names(self) -> Locator:
        return self.page.locator("[id^=hotel-card-name-]")

    def search(self, query: str) -> None:
        with allure.step(f"Ищем отель: «{query}»"):
            self.search_input.fill(query)
